system-preparation/2ajf_ace2/simulate_2ajf_ace2.py

```diff
@@ -0,0 +1,106 @@
+
+# Add hydrogens and solvate
+print('Adding hydrogens and solvent...')
+modeller = app.Modeller(pdb.topology, pdb.positions)
+# modeller.addHydrogens(forcefield)
+modeller.addSolvent(forcefield, model=water_model, padding=solvent_padding, ionicStrength=ionic_strength)
+print('System has %d atoms' % modeller.topology.getNumAtoms())
+
+# Write initial model
+print('Writing initial solvated system to %s' % solvated_pdb_filename)
+with open(output_prefix + solvated_pdb_filename, 'w') as outfile:
+    app.PDBFile.writeFile(modeller.topology, modeller.positions, file=outfile, keepIds=True)
+
+# Create the system
+print('Creating OpenMM System...')
+system = forcefield.createSystem(modeller.topology, nonbondedMethod=app.PME, constraints=app.HBonds, removeCMMotion=False, hydrogenMass=hydrogen_mass)
+
+# Add a barostat
+print('Adding barostat...')
+barostat = openmm.MonteCarloBarostat(pressure, temperature)
+system.addForce(barostat)
+
+# Serialize integrator
+print('Serializing integrator to %s' % integrator_xml_filename)
+integrator = openmm.LangevinIntegrator(temperature, collision_rate, timestep)
+with open(output_prefix + integrator_xml_filename, 'w') as outfile:
+    xml = openmm.XmlSerializer.serialize(integrator)
+    outfile.write(xml)
+
+# Minimize
+print('Minimizing energy...')
+context = openmm.Context(system, integrator)
+context.setPositions(modeller.positions)
+print('  initial : %8.3f kcal/mol' % (context.getState(getEnergy=True).getPotentialEnergy()/unit.kilocalories_per_mole))
+openmm.LocalEnergyMinimizer.minimize(context)
+print('  final   : %8.3f kcal/mol' % (context.getState(getEnergy=True).getPotentialEnergy()/unit.kilocalories_per_mole))
+with open(output_prefix + minimized_pdb_filename, 'w') as outfile:
+    app.PDBFile.writeFile(modeller.topology, context.getState(getPositions=True,enforcePeriodicBox=True).getPositions(), file=outfile, keepIds=True)
+
+# Equilibrate
+print('Equilibrating...')
+initial_time = time.time()
+for iteration in progressbar.progressbar(range(niterations)):
+    integrator.step(nsteps)
+elapsed_time = (time.time() - initial_time) * unit.seconds
+simulation_time = niterations * nsteps * timestep
+print('    Equilibration took %.3f s for %.3f ns (%8.3f ns/day)' % (elapsed_time / unit.seconds, simulation_time / unit.nanoseconds, simulation_time / elapsed_time * unit.day / unit.nanoseconds))
+with open(output_prefix + equilibrated_pdb_filename, 'w') as outfile:
+    app.PDBFile.writeFile(modeller.topology, context.getState(getPositions=True,enforcePeriodicBox=True).getPositions(), file=outfile, keepIds=True)
+print('  final   : %8.3f kcal/mol' % (context.getState(getEnergy=True).getPotentialEnergy()/unit.kilocalories_per_mole))
+
+# Serialize state
+print('Serializing state to %s' % state_xml_filename)
+state = context.getState(getPositions=True, getVelocities=True, getEnergy=True, getForces=True)
+with open(output_prefix + state_xml_filename, 'w') as outfile:
+    xml = openmm.XmlSerializer.serialize(state)
```

Comment:
If I'm right this kind of code could be factorized into a dedicated function, since you reimplement the same
treatment in this script.

It could be something like:

```python
def write_output(filename, state):
    with open(os.path.join(output_prefix, filename), 'w') as outfile:
        xml = openmm.XmlSerializer.serialize(state)
        outfile.write(xml)
```

And then you juste have to call it like this during the execution:
```
write_output(state_xml_filename, state)
```